File: GO_단계별로풀어보기/04.1차원배열/08.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var input, remainder int
	var remainders []int

	for {
		_, err := fmt.Fscan(reader, &input)
		if err != nil {
			break
		}
		remainder = input % 42
		remainders = append(remainders, remainder)
	}
	res := removeDuplicateInt(remainders)
	fmt.Fprint(writer, len(res))
}

func removeDuplicateInt(intSlice []int) []int {
	allKeys := make(map[int]bool)
	slice := []int{}
	for _, item := range intSlice {
		if _, value := allKeys[item]; !value {
			allKeys[item] = true
			slice = append(slice, item)
		}
	}
	return slice
}

File: 단계별로풀어보기/14.집합과맵/08.py
# 백준 결과: '틀렸습니다'
# 반례 못 찾음

import sys

input = sys.stdin.readline
print = sys.stdout.write

string = input().rstrip("\n")
len_string = len(string)
res = 0

for len_sub_string in range(1, len_string + 1):
    sub_strings = set(
        string[p : p + len_sub_string] 
        for p in range(len_string - len_sub_string + 1)
    )
    res += len(sub_strings)

print(str(res))
